src/mlpack/methods/isomap/mds.hpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file mds.hpp
+ * @author Rishabh Ranjan
+ * 
+ * This file defines MDS class which implements classical Multidimensional
+ * Scaling which is a dimensionality reduction algorithm which uses spectral
+ * decomposition of dissimilarity matix.
+ * This implementation is based on MATLAB's implementation of classical
+ * multidimensional scaling (file - cmdscale.m).
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ISOMAP_MDS
+#define MLPACK_METHODS_ISOMAP_MDS
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace isomap {
+
+/**
+ * This class implements classical Multidimensional Scaling.
+ */
+class MDS
+{
+ public:
+  /**
+   * Fucntion to implement classical Multidimensional Scaling on a given
+   * dissimilarity matrix.
+   * 
+   * @param disMat -dissimilarity matrix (distance matrix)
+   */
+  void Apply(arma::mat& disMat)
+  {
+    // check to see if the incoming matrix is valid for cMDS
+    IsValidMatrix(disMat);
```

Comment:
This check does take some amount of time to perform.  I would be okay if we omitted it and instead just required it in the documentation.  We are already sure that `Isomap` will be passing us a square and symmetric dissimilarity matrix.

Reply:
I put this check thinking MDS could be used separately. I think it's better to have a separate mds/ for this file as you suggested. I will do that.

Reply:
I removed the this check and included in the `PROGRAM_INFO` that if the input to cMDS is dissimilarity matrix, then it should be symmetric, since this would save some time.